xds/src/main/java/io/grpc/xds/RingHashLoadBalancer.java

```diff
@@ -334,23 +339,30 @@ public static EquivalentAddressGroup stripAttrs(EquivalentAddressGroup eag) {
   }
 
   private static final class RingHashPicker extends SubchannelPicker {
+    private final Random random = new Random();
```

Comment:
We probably want to share this object across pickers and even inject it when creating RingHashLB for tests. That way we don't have to deal with flakes.